ComponentKit/DataSources/CKCollectionViewDataSource.mm

```diff
@@ -73,6 +73,11 @@ - (instancetype)init
   CK_NOT_DESIGNATED_INITIALIZER();
 }
 
+- (void)dealloc
+{
+    _collectionView.dataSource = nil;
```

Comment:
> Prior iOS9 UIKit does not use ARC, we have to clear weak ref manually.

Ah, I see--sorry I missed this! Is this really the case, though? Can you link me to some documentation? I'm investigating myself as well.